File: ps_java_kotlin/ps_java/boj/src/main/java/queue/BOJ10866.java
package queue;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class BOJ10866 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String command;
        StringBuilder sb = new StringBuilder();

        Deque<Integer> deque = new ArrayDeque<>();
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            command = st.nextToken();
            switch (command) {
                case "push_front":
                    deque.offerFirst(Integer.parseInt(st.nextToken()));
                    break;
                case "push_back":
                    deque.offerLast(Integer.parseInt(st.nextToken()));
                    break;
                case "pop_front":
                    if (deque.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(deque.pollFirst()).append("\n");
                    break;
                case "pop_back":
                    if (deque.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(deque.pollLast()).append("\n");
                    break;
                case "size":
                    sb.append(deque.size()).append("\n");
                    break;
                case "empty":
                    if (deque.isEmpty())
                        sb.append("1\n");
                    else
                        sb.append("0\n");
                    break;
                case "front":
                    if (deque.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(deque.getFirst()).append("\n");
                    break;
                case "back":
                    if (deque.isEmpty())
                        sb.append("-1\n");
                    else
                        sb.append(deque.getLast()).append("\n");
                    break;
            }
        }

        System.out.println(sb.toString());
    }
}

File: sample/ps_kotlin/boj/src/main/kotlin/boj10866.kt
fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()

    val q = ArrayDeque<String>()

    val n = br.readLine().toInt()
    repeat(n) {
        val command = br.readLine()
        when {
            command.startsWith("push_front") -> {
                val value = command.split(" ")[1]
                q.addFirst(value)
            }
            command.startsWith("push_back") -> {
                val value = command.split(" ")[1]
                q.addLast(value)
            }
            command == "front" -> {
                if (q.isEmpty()) bw.write("-1\n") else bw.write("${q.first()}\n")
            }
            command == "back" -> {
                if (q.isEmpty()) bw.write("-1\n") else bw.write("${q.last()}\n")
            }
            command == "size" -> {
                bw.write("${q.size}\n")
            }
            command == "empty" -> {
                if (q.isEmpty()) bw.write("1\n") else bw.write("0\n")
            }
            command == "pop_front" -> {
                if (q.isEmpty()) bw.write("-1\n") else bw.write("${q.removeFirst()}\n")
            }
            command == "pop_back" -> {
                if (q.isEmpty()) bw.write("-1\n") else bw.write("${q.removeLast()}\n")
            }
        }
    }
    bw.flush()
}
